numpy/core/src/multiarray/textreading/rows.c

```diff
@@ -179,20 +274,14 @@ read_rows(stream *s,
     /* We give a warning if max_rows is used and an empty line is encountered */
     bool give_empty_row_warning = max_rows >= 0;
 
-    int ts_result = 0;
-    tokenizer_state ts;
-    if (tokenizer_init(&ts, pconfig) < 0) {
-        goto error;
-    }
-
     /* Set the actual number of fields if it is already known, otherwise -1 */
     Py_ssize_t actual_num_fields = -1;
-    if (usecols != NULL) {
+    if (usecols_arr != NULL) {
         assert(homogeneous || num_field_types == num_usecols);
         actual_num_fields = num_usecols;
     }
     else if (!homogeneous) {
-        assert(usecols == NULL || num_field_types == num_usecols);
+        assert(num_field_types == num_usecols);
```

Comment:
This assert is failing (if you are looking for that).  `num_usecols` is not necessarily defined here, is it?

Reply:
Ah, the result of an incomplete clean up.  I just removed the assert().